README.md

```diff
@@ -0,0 +1,46 @@
+# paascharmgen
+
+`paascharmgen` is a tool to generate Go structs for a 12 factor Go app generated using rockcraft and charmcraft
+go-framework extensions.
+
+
+## Usage
+
+Create a Kubernetes charm for a Go project following the [tutorial (TODO)](https://juju.is/docs/sdk/write-your-first-kubernetes-charm-for-a-go-app) using
+the rockcraft and charmcraft go-framework.
+
+You can choose to install the tool with `go install github.com/canonical/paascharmgen@latest` or run it directly without install.
+
+You can add a directive similar to the following one to one of your Go files, replacing the placeholders:
+`//go:generate paascharmgen -c [CHARMCRAFT_YAML_DIRECTORY] -o [OUTPUT_GO_FILE] -p [PACKAGE_NAME]`
+
+Without installing, you can create a directive like the following one:
+`//go:generate go run github.com/canonical/paascharmgen@latest -c [CHARMCRAFT_YAML_DIRECTORY] -o [OUTPUT_GO_FILE] -p [PACKAGE_NAME]`
+
+You can also use the tool directly without a `go:generate` directive. Once the Go source code file with the structs is generated,
+you can use it with the "github.com/caarlos0/env/v11" module like:
```

Comment:
What is `caarlos0` please?

Reply:
It's a library used to map envvars to structs, basically any of this type of library can be used. Maybe provide example with the most common, or if we provide one, it should be the one we consider "the best"

Reply:
This is the library suggested in the spec. If we do not want to write our env to structs  library we will need to use one of those libraries (this library is probably the most popular that is maintained).